### PR TITLE
coprocessor/chunk: add comments and inline for chunk

### PR DESCRIPTION
**Pull request by siddontang**
## What have you changed? (mandatory)

1. Add missing docs for coprocessor chunk. 
2. Add inline for some short functions.

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

No need test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Add a few positive/negative examples (optional)

/cc @AndreMouche @breeswish 

PTAL @queenypingcap

### CODE REVIEW

File: src/coprocessor/codec/chunk/chunk.rs [hunk 1]
```diff
@@ -60,11 +63,14 @@ impl Chunk {
         }
     }
 
+    /// Append a dataum to the column
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/chunk.rs]: dataum -> datum

File: src/coprocessor/codec/chunk/chunk.rs [hunk 2]
```diff
@@ -112,20 +121,25 @@ impl<'a> Row<'a> {
...
         self.idx
     }
 
     /// Get the number of values in the row.
+    #[inline]
     pub fn len(&self) -> usize {
         self.c.num_cols()
     }
 
+    /// Get the datum of column with specified type in the row.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/chunk.rs]: "Get the datum of **a** column with **the** specified type in the row."
OR 
"Get the datum of **the** column with **the** specified type in the row."?

File: src/coprocessor/codec/chunk/column.rs [hunk 3]
```diff
@@ -35,6 +36,7 @@ pub struct Column {
 }
 
 impl Column {
+    /// Create the column with specified type and capacity.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: Create the column with **a** specified type and capacity.

File: src/coprocessor/codec/chunk/column.rs [hunk 4]
```diff
@@ -56,6 +58,7 @@ impl Column {
         }
     }
 
+    /// Get the datum of one row with specified type.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: Get the datum of one row with **the** specified type.

File: src/coprocessor/codec/chunk/column.rs [hunk 5]
```diff
@@ -111,6 +115,7 @@ impl Column {
         }
     }
 
+    /// Create a column which element's length is fixed.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: -> Create a column with a fixed element length.

File: src/coprocessor/codec/chunk/column.rs [hunk 6]
```diff
@@ -120,6 +125,7 @@ impl Column {
         }
     }
 
+    /// Create a column which element's length is variant.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: -> Create a column with variant element length.

File: src/coprocessor/codec/chunk/column.rs [hunk 7]
```diff
@@ -131,11 +137,13 @@ impl Column {
         }
     }
 
+    /// Return whether the column has a fixed length element or not.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: -> Return whether the column has a fixed element length or not.

File: src/coprocessor/codec/chunk/column.rs [hunk 8]
```diff
@@ -181,49 +192,57 @@ impl Column {
         Ok(())
     }
 
+    /// Called when finish appending fixed datum.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: -> Called when the fixed datum has been appended.

File: src/coprocessor/codec/chunk/column.rs [hunk 9]
```diff
@@ -181,49 +192,57 @@ impl Column {
...
 
+    /// Get the f64 datum of the row in the column.
     pub fn get_f64(&self, idx: usize) -> Result<f64> {
         let start = idx * self.fixed_len;
         let end = start + self.fixed_len;
         let mut data = &self.data[start..end];
         number::decode_f64_le(&mut data).map_err(Error::from)
     }
 
+    /// Called when finish appending variant datum.
```
Comment (CaitinChen) [inline src/coprocessor/codec/chunk/column.rs]: -> Called when the variant datum has been appended.

**Comment by siddontang** [comment]
PTAL @AndreMouche @CaitinChen

**Pull request merged**